src/algorithms/crypto/randomness.md

```diff
@@ -10,54 +10,49 @@ We describe this formatting below.
-- **Objects**: Their specified Serialization, currently CBOR-based serialization defined on algebraic datatypes
+Entropy from the ticket-chain can be combined with other values to generate necessary randomness that can be
+specific to (eg) a given miner address or epoch. To be used as part of entropy, these values are combined in 
+objects that can then be CBOR-serialized according to their algebraic datatypes.
 
 ## Domain Separation Tags
 
-For {{<sref crypto_signatures>}} as well as {{<sref vrf>}} usage in the protocol, we define Domain Separation Tags with which we prepend random inputs.
-
-The source of truth is defined below, but the currently defined DSTs are:
+Further, we define Domain Separation Tags with which we prepend random inputs when creating entropy.
 
-- for drawing randomness from an on-chain ticket:
-    - `TicketDrawingDST = 1`
-- for generating a new random ticket:
-    - `TicketProductionDST = 2`
-- for generating randomness for running ElectionPoSt:
-    - `ElectionPoStChallengeSeedDST = 3`
-- for generating randomness for running SurprisePoSt:
-    - `SurprisePoStChallengeSeedDST = 4`
-- for selection of which miners to surprise:
-    - `SurprisePoStSelectMinersDST = 5`
-- for selection of which sectors to sample:
-	- `SurprisePoStSampleSectors = 6`
+All randomness used in the protocol must be generated in conjunction with a unique DST, as well as 
+certain {{<sref crypto_signatures>}} and {{<sref vrf>}} usage.
 
 ## Forming Randomness Seeds
 
 Different uses of randomness require randomness seeds predicated on a variety of inputs. For instance, we have:
 
-- `TicketDrawing` -- uses ticket and epoch number
 - `TicketProduction` -- uses ticket and miner actor addr
-- `PoStChallengeSeed` -- uses ticket and miner actor addr
-- `SurprisePoStSelectMiners` -- uses ticket and epoch number
+- `ElectionPoStChallengeSeed` -- uses ticket and miner actor addr
+- `WindowedPoStChallengeSeed` -- uses ticket and epoch number
 ...
 
 In all cases, a ticket is used as the base of randomness (see {{<sref tickets>}}). In order to make randomness seed creation uniform, the protocol derives all such seeds in the same way, as follows (also see {{<sref tickets>}}):
+
+For a given randomness lookback `l`, and serialized entropy `s`:
+
 ```text
- For a given ticket's randomness ticket_randomness:
+ticket = draw_ticket_from_chain(l)
+ticket_randomness = ticket.digest
 
 buffer = Bytes{}
 buffer.append(IntToBigEndianBytes(AppropriateDST))
-buffer.append(-1) // a flag to be used in cases where FIL might need longer randomness outputs. Currently unused
 buffer.append(ticket_randomness)
-buffer.append(other needed serialized inputs)
+buffer.append(s)
 
-randomness = SHA256(buffer)
+randomness = H(buffer)
 ```
 
 {{< readfile file="/docs/actors/actors/crypto/randomness.go" code="true" lang="go" >}}
 {{< readfile file="/docs/systems/filecoin_blockchain/struct/chain/chain.go" code="true" lang="go" >}}
+
+## Drawing randomness prior to genesis
```

Comment:
This section will be immensely clearer and more likely to be correct if you specify this special case in the already defined framework built up above.  You should specify that pre-genesis special cases all derive entropy bytes from a certain form.  The below section doesn't give enough detail for the implementer to understand how all types of pre-genesis drawing randomness work.  For example what if you want to draw with an epoch and miner address?  Do you cbor serialize curr - lookback in the epoch field of the entropy struct?  Do you serialize the epoch bytes of curr - lookback separately and fill the entropy struct with the epoch passed in?  By defining serialization separately here you also erase mention of domain separation tags which you probably don't want to do.

Reply:
Lmk if new version looks good. Thanks for this feedback. v useful!